server/test/services/rflink/controllers/rflink.controller.test.js

```diff
@@ -0,0 +1,291 @@
+const sinon = require('sinon');
+
+const { assert, fake, stub } = sinon;
+const { expect } = require('chai');
+const RFLinkController = require('../../../../services/rflink/api/rflink.controller');
+const { ServiceNotConfiguredError } = require('../../../../utils/coreErrors');
+
+const serviceId = '6d1bd783-ab5c-4d90-8551-6bc5fcd02212';
+const gladys = {
+  service: {
+    getLocalServiceByName: stub().resolves({
+      id: serviceId,
+    }),
+  },
+  event: {
+    emit: fake.returns(null),
+  },
+  stateManager: {
+    get: fake.returns(null),
+  },
+  variable: {
+    getValue: stub()
+      .withArgs('RFLINK_PATH', serviceId)
+      .resolves('//tty'),
+  },
+};
+
+const rflinkHandler = {
+  gladys: fake.returns(gladys),
+  serviceId: fake.returns(serviceId),
+  connected: fake.returns(false),
+  ready: fake.returns(false),
+  scanInProgress: fake.returns(false),
+  newDevices: [],
+  devices: fake.returns([]),
+  currentMilightGateway: fake.returns('F746'),
+  milightBridges: fake.returns({}),
+  pair: stub().resolves(),
+  unpair: stub().resolves(),
+  getNewDevices: fake.resolves(true),
+  connect: stub()
+    .withArgs('rflinkPath')
+    .resolves('connected'),
+  disconnect: stub().resolves(),
+  sendUsb: {
+    write: stub()
+      .withArgs('command')
+      .resolves(),
+  },
+};
+
+describe('POST /api/v1/service/rflink/connect', () => {
+  let controller;
+
+  beforeEach(() => {
+    controller = RFLinkController(gladys, rflinkHandler, serviceId);
+  });
+
+  afterEach(() => {
+    sinon.reset();
+  });
+
+  it('should connect successfully', async () => {
+    const req = {};
+    const res = {
+      json: fake.returns(null),
+    };
+    await controller['post /api/v1/service/rflink/connect'].controller(req, res);
+    assert.calledOnce(gladys.variable.getValue);
+    // @Todo assert.calledOnce(rflinkHandler.connect);
+    // @Todo assert.calledOnce(res.json);
+  });
+
+  it('should raise an error on connection when no path is given', async () => {
+    gladys.variable.getValue = stub()
+      .withArgs('RFLINK_PATH')
+      .resolves(undefined);
+    controller = RFLinkController(gladys, rflinkHandler);
+
+    const req = {};
+    const res = { json: fake.returns(true) };
+    try {
+      await controller['post /api/v1/service/rflink/connect'].controller(req, res);
+    } catch (e) {
+      expect(e).to.be.instanceOf(ServiceNotConfiguredError);
+    }
+    // @Todo assert.calledOnce(rflinkHandler.connect);
+    // @Todo assert.calledOnce(res.json); KO for unknown reason :(
```

Comment:
This warns me... todos in test... it tells me you miss something.

Reply:
Tests were improved (I am still not as comfortable with tests)